python/cuml/cuml/common/sparsefuncs.py

```diff
@@ -375,6 +376,7 @@ def extract_knn_graph(knn_info, n_neighbors):
         deepcopy=deepcopy,
         check_dtype=np.float32,
         convert_to_dtype=np.float32,
+        convert_to_mem_type=mem_type,
```

Comment:
If this parameter is not specified, the default behavior is to copy to GPU?

Reply:
Explanation of this argument: https://github.com/rapidsai/cuml/blob/ccf23dcce8f94918bed2be8c70c3fd3384a6e991/python/cuml/cuml/internals/array.py#L964-L968

Default value for this argument is `None`, which means we convert the array to `cuml.global_settings.memory_type` (which I believe defaults to device memory)!